.github/workflows/quality_check.yml

```
@@ -46,7 +46,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -r requirements.txt
+          find . -type f -name "requirements.txt" | sed -e 's/^/-r /' | xargs pip install
```

Also manually install pytest (just to be safe)

```suggestion
          pip install pytest
          find . -type f -name "requirements.txt" | sed -e 's/^/-r /' | xargs pip install
```